src/store/aggregator/sagas/transferTokens.test.ts

```diff
@@ -0,0 +1,347 @@
+          [matchers.call.fn(mockToken.allowance), parseUnits('0')],
+          [matchers.call.fn(mockToken.approve), mockExtendTx],
+          [matchers.call.fn(mockBridge.receiveTokensAt), mockExtendTx],
+          [matchers.call.fn(mockExtendTx.wait), mockExtendReceipt],
+        ])
+        .hasFinalState(expectedState)
+        .run();
+    });
+
+    it('when no address provided', async () => {
+      const expectedState: AggregatorState = {
+        ...aggregatorInitialState,
+        submitCall: {
+          status: 'failure',
+          steps: [
+            {
+              ...initialSteps?.[0],
+              error: 'Could not find addresses',
+            },
+            ...initialSteps.slice(1, 4),
+          ],
+        },
+      };
+
+      await expectSaga(
+        transferTokens,
+        aggregatorActions.submit(depositMockValues)
+      )
+        .withReducer(aggregatorReducer)
+        .withState(aggregatorInitialState)
+        .provide([
+          ...mockSelectors.slice(0, -1),
+          [matchers.select(accountSelector), undefined],
+          [matchers.call.fn(mockToken.allowance), parseUnits('15')],
+          [matchers.call.fn(mockToken.approve), mockExtendTx],
+          [matchers.call.fn(mockBridge.receiveTokensAt), mockExtendTx],
+          [matchers.call.fn(mockExtendTx.wait), mockExtendReceipt],
+        ])
+        .hasFinalState(expectedState)
+        .run();
+    });
+  });
+
+  describe('withdraw', () => {
+    const mockSelectors: (EffectProviders | StaticProvider)[] = [
+      [matchers.select(flowStateSelector), 'withdraw'],
+      [matchers.select(startingTokenDecimalsSelector), mockTokenDecimals],
+      [matchers.select(startingTokenContractSelector), mockToken],
+      [matchers.select(bassetAddressSelector), mockBassetAddress],
+      [matchers.select(accountSelector), mockAccount],
+      [matchers.select(massetContractSelector), mockMasset],
+    ];
+
+    it('when sufficient allowance', async () => {
+      const expectedState: AggregatorState = {
+        ...aggregatorInitialState,
+        submitCall: {
+          status: 'success',
+          currentOperation: 'withdraw',
+          steps: [
+            {
+              ...initialSteps?.[3],
+              tx: mockExtendTx,
+              txReceipt: mockExtendReceipt,
+            },
+          ],
+        },
+      };
+
+      await expectSaga(
+        transferTokens,
+        aggregatorActions.submit(depositMockValues)
+      )
+        .withReducer(aggregatorReducer)
+        .withState(aggregatorInitialState)
+        .provide([
+          ...mockSelectors,
+          [matchers.call.fn(mockToken.allowance), parseUnits('11')],
+          [
+            matchers.call.fn(
+              mockMasset['redeemToBridge(address,uint256,address)']
+            ),
+            mockExtendTx,
+          ],
+          [matchers.call.fn(mockExtendTx.wait), mockExtendReceipt],
+        ])
+        .hasFinalState(expectedState)
```

Comment:
add more checks to verify that proper effects are yielded(For me in that case the most important ones are: `.call`)